### PR TITLE
test: ability to select cluster template with MachinePool for custom K8s build

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

 

 
**What type of PR is this?**

**What this PR does / why we need it**:

Follow-up PR for https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1419. `./scripts/ci-entrypoint.sh` will select the cluster template with MachinePool for custom K8s build if `EXP_MACHINE_POOL` is set to `true`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -417,15 +417,16 @@ To run a custom test suite on a CAPZ cluster locally, set `AZURE_CLIENT_ID`, `AZ
...
-| `KUBECONFIG`                   | Provide your existing cluster kubeconfig filepath. If no kubeconfig is provided, `./kubeconfig` will be used.     |
-| `USE_CI_ARTIFACTS`             | Use a CI version of Kubernetes, ie. not a released version (eg. `v1.19.0-alpha.1.426+0926c9c47677e9`)         |
-| `CI_VERSION`                   | Provide a custom CI version of Kubernetes. By default, the latest master commit will be used.                 |
-| `EXP_MACHINE_POOL`             | Use [Machine Pool](topics/machinepools.md) for worker machines.                                               |
+| Variable                       | Description                                                                                                                                |
+|--------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------|
+| `AZURE_SSH_PUBLIC_KEY_FILE`    | Use your own SSH key.                                                                                                                      |
```
Comment (CecileRobertMichon) [inline docs/development.md:422]: can you move this to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/docs/book/src/developers/development.md ? I need to delete this file again, looks like it got re-added in a rebase

Comment (chewong) [reply docs/development.md:422]: Deleted in this PR

File: scripts/ci-entrypoint.sh [hunk 2]
```diff
@@ -88,7 +88,11 @@ select_cluster_template() {
     fi
 
     if [[ "${EXP_MACHINE_POOL:-}" == "true" ]]; then
-        export CLUSTER_TEMPLATE="${CLUSTER_TEMPLATE/prow/prow-machine-pool}"
+        if [[ "${CLUSTER_TEMPLATE}" =~ "prow" ]]; then
+            export CLUSTER_TEMPLATE="${CLUSTER_TEMPLATE/prow/prow-machine-pool}"
+        elif [[ "${CLUSTER_TEMPLATE}" =~ "custom-builds" ]]; then
```
Comment (CecileRobertMichon) [inline scripts/ci-entrypoint.sh:93]: we'll need to deal with CCM + machine pool too at some point

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

File: scripts/ci-entrypoint.sh (hunk 2 above)
Comment (chewong) [reply scripts/ci-entrypoint.sh:93]: Let's refactor the script in the near future. We should also address https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1372 at the same time.

**Pull request merged**